### DISCUSSION TITLE
package.json inheritance

### DISCUSSION THREAD

**Discussion by spencer17x** [pnpm/pnpm#5430] [Ideas]
In monorepo，There may be multiple packages that contain the same dependency or script.
Can I extend package.json by extending the fields of package.json so that I can inherit from the parent class's package.json.
I saw that someone from the repo of NPM raised the relevant issue, but NPM does not support it. Is there any way for PNPM to support it, or does it already support it.

related links:
* https://github.com/pnpm/pnpm/issues/1813
* https://stackoverflow.com/questions/42441543/npm-package-json-inheritance
* https://github.com/npm/npm/issues/8112

**Comment by Cpt-Falcon** [comment]
What happened with this?

**Comment by spock123** [comment]
I don't think this would work, as most tools would not be able to work with it, as they don't know the "extends" field.

What could work is a tool to alter package.json files, and have that tool run on "postInstall" for example. Tools like
https://github.com/microsoft/package-inherit
